### PR TITLE
Add Babylon.JS

### PR DESCRIPTION
**Pull request by diedyesterdaywashere**
### SteamDB app page links to a few games using this
https://steamdb.info/app/842490/

### Brief explanation of the change

Added rule for Babylon.JS

### CODE REVIEW

File: rules.ini [hunk 1]
```diff
@@ -37,6 +37,7 @@ AdventureGameStudio = (?:^|/)(?:AGSteam\.dll|acsetup\.cfg)$
 AGI = (?:^|/)AGIDATA\.OVL$
 Amazon_Lumberyard = ^bootstrap\.cfg$
 Asura = \.asr$
+BabylonJS = babylon
```
Comment (Lifeismana) [inline rules.ini:40]: is this missing a $ at the end ? (looking at the examples you gave, this might be it)

**Comment by Lifeismana** [comment]
regex is way too "wide" but it might be due to the missing $
 
* `babylon.js` - 57 matches
* `babylon` - 21 matches
* `civ5buildings_babylon.xml` - 8 matches
* `texture_loyalty_overlay_babylon` - 6 matches
* `babylon.civ5pkg` - 5 matches
* `station_newbabylon.lua` - 5 matches
* `hanging_gardens_of_babylon.item` - 4 matches
* `flagdecal_babylon.dds` - 4 matches
* `babylonian_817.bin` - 4 matches
* `technobabylon.exe` - 4 matches
* `11. building - babylon.png` - 4 matches
* `scenerybabylon.arc` - 3 matches
* `babylonia.txt` - 3 matches
* `nbabylonb1.xml` - 3 matches
* `babylon_station.mtl` - 3 matches
* `boss_magma_babylon` - 3 matches
* `04-ael babylonia.luxb` - 3 matches
* `babylons_fall.exe` - 3 matches
* `#&plan_babylon.tga.dds` - 3 matches
* `hanging_gardens_of_babylon.txt` - 2 matches
* `be-002-babylon.obj` - 2 matches
* `214 - babylon.txt` - 2 matches
* `babylon_arch.obj` - 2 matches
* `babylon.ogg` - 2 matches
* `rw_babylon.mmm` - 2 matches
* `babylonfall01s1a.mid` - 2 matches
* `babylon_i.xml` - 2 matches
* `spine_babylon_a9fba079469e3a84ebcd6467e9d3abba` - 2 matches
* `babylon.bank` - 2 matches
* `05_buildings_babylon.pal` - 2 matches
* `ricky babylon.png` - 2 matches
* `st_babylon.ogg` - 2 matches
* `readme.txt` - 2 matches
* `parser-babylon.js` - 2 matches
* `babylonjs` - 2 matches
* `pi_babylon` - 2 matches
* `mod_babylon.js` - 2 matches

File: rules.ini (hunk 1 above)
Comment (diedyesterdaywashere) [reply rules.ini:40]: yeye i think i just forgot the $, sorry

**Comment by diedyesterdaywashere** [comment]
changed a logic of the tests since babylon.js got most hits

**Comment by diedyesterdaywashere** [comment]
but i guess i can do another thing

File: tests/types/_NonMatchingTests.txt [hunk 2]
```diff
@@ -116,6 +122,8 @@ data.dcpcp
 testxem
 .xemtest
 fdata.dcp
+babylo
+abylon
```
Comment (Lifeismana) [inline tests/types/_NonMatchingTests.txt:126]:
```suggestion
```

File: tests/types/_NonMatchingTests.txt [hunk 3]
```diff
@@ -15,6 +15,12 @@ fna.dllwhoops
 sub/dir/notactuallyfna.dll
 sub/dir/fna.dllwhoops
 fx_core_dll
+abylon.js
+babylon.j
+babylon_js
+lib_babylon
+notrelatedto.babylo
+notrelatedto_babylon
```
Comment (Lifeismana) [inline tests/types/_NonMatchingTests.txt:23]:
```suggestion
```

File: rules.ini [hunk 4]
```diff
@@ -37,6 +37,7 @@ AdventureGameStudio = (?:^|/)(?:AGSteam\.dll|acsetup\.cfg)$
 AGI = (?:^|/)AGIDATA\.OVL$
 Amazon_Lumberyard = ^bootstrap\.cfg$
 Asura = \.asr$
+BabylonJS = (?:^|/)(?:babylon\.js|\.babylon|lib/babylon)$
```
Comment (Lifeismana) [inline rules.ini:40]:
```suggestion
BabylonJS = (?:^|/)(?:babylon\.js|lib/babylon)$
```
`lib/babylon` is only for one game, so i'm not sure if it's worth it to have it

File: tests/types/Engine.BabylonJS.txt
Comment (Lifeismana) [inline tests/types/Engine.BabylonJS.txt:1]: needs a cleanup of the removed rule and this string can be added:
```
www/js/libs/babylon.js
```

**Review by Lifeismana — CHANGES_REQUESTED** [review]

**Pull request merged**